lib/vanagon/project.rb

```diff
@@ -270,14 +270,25 @@ def has_provides?
       !get_provides.empty?
     end
 
-    # Collects the preinstall packaging actions for the project and it's components
-    # for the specified packaging state
+    # Checks that the pkg_state is valid, return vanagon error if invalid
     #
-    # @param pkg_state [String] the package state we want to run the given scripts for.
+    # @param String pkg_state String of package state input
+    # @param String function the function that called
+    def check_pkg_state(pkg_state, method_name)
+      if pkg_state.empty? || !(["install", "upgrade"].include? pkg_state)
```

Comment:
Is `!(["install", "upgrade"].include? pkg_state)` ultimately just comparing String values? I've probably done it that way in Vanagon before too but I think maybe I regret it if I did? Anyway, if this is just a String comparison, then you could simplify it to

```ruby
# case insensitive, because why not?
pkg_state.match? /^upgrade|install$/i
```

Reply:
`pkg_state` is an array (that should contain one or both of 'upgrade' and 'install'), so that won't work.

Reply:
If `pkg_state` is an Array, then what's there now has some edge cases.

```ruby
>> ["install", "upgrade"].include? pkg_state
=> true
>> pkg_state
=> "upgrade"
>> pkg_state = %w[upgrade]
=> [
  [0] "upgrade"
]
>> pkg_state
=> [
  [0] "upgrade"
]
>> ["install", "upgrade"].include? pkg_state
=> false
>>
```
Will pkg_state ever contain anything but the Strings "install" and "upgrade"?